src/Microsoft.AspNetCore.Mvc.TagHelpers/Internal/CacheTagHelperMemoryCacheFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers.Internal
+{
+    public class CacheTagHelperMemoryCacheFactory
```

Comment:
Let me know if you want unit tests for this. I believe it's ok as it's covered by functional tests.